pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -653,6 +653,9 @@ type ThanosSpec struct {
 	LogFormat string `json:"logFormat,omitempty"`
 	// MinTime for Thanos sidecar to be configured with. Option can be a constant time in RFC3339 format or time duration relative to current time, such as -1d or 2h45m. Valid duration units are ms, s, m, h, d, w, y.
 	MinTime string `json:"minTime,omitempty"`
+	// BlockSize controls the size of TSDB blocks produced by Prometheus. Default is 2h to match the upstream Prometheus defaults.
+	// WARNING: Changing this can impact the performance and efficiency of the entire Prometheus/Thanos stack due to how it interacts with memory and Thanos compactors. It is recommended to keep this value set to a multiple of 120 times your longest scrape or rule interval. For example, 30s * 120 = 1h.
+	BlockSize Duration `json:"blockSize,omitempty"`
```

Comment:
```suggestion
	// +kubebuilder:default:="2h"
	BlockSize Duration `json:"blockSize,omitempty"`
```

Reply:
(nit) Should it be named `BlockDuration` rather than `BlockSize`?

Reply:
Yea, Duration is probably better.